modules/channel-telegram/src/senders/typing.ts

```diff
@@ -0,0 +1,27 @@
+import { ChannelSender } from 'common/channel'
+import { TelegramContext } from '../backend/typings'
+
+export class TelegramTypingSender implements ChannelSender<TelegramContext> {
+  get channel(): string {
+    return 'telegram'
+  }
+
+  get priority(): number {
+    return -1
+  }
+
+  get id() {
+    return TelegramTypingSender.name
+  }
+
+  handles(context: TelegramContext): boolean {
+    const typing = context.event.payload.typing
+    return context.handlers?.length > 0 && (typing === undefined || typing === true)
+  }
+
+  async send(context: TelegramContext) {
+    const delay = context.event.payload.delay ?? 1000
+    await context.client.telegram.sendChatAction(context.chatId, 'typing')
+    await Promise.delay(delay)
```

Comment:
Can you validate this is still relevant?

Reply:
Mm payload.delay isn't something we can set anywhere I think at the moment (in the ui at least). It doesn't really matter though because we have default behaviour if it's not set.

As for the Promise.delay it seems alright to me. I don't think we've had any problem using it so far. If we decide to remove these latter on we can do it in another PR.